### ISSUE TITLE
Punctuation in Greek

### ISSUE THREAD

**Issue by mnjames** [mimbrown/render#56]
Solving this will almost certainly require handling the underlying, raw data differently. As a result we might need to wait until after the trial publication (and I think it's fine if we do).

Note that it is likely that we'll need to deal with this issue before #44 .

The Raw_Interlinear.xml data we're currently using is not the actual text of Sinaiticus--it's data regarding the matching of Greek with Urdu glosses. It contains almost all the data of Sinaiticus, but not the punctuation. It also wouldn't contain footnotes if we ever added them into Sinaiticus (see #44).

The actual data of Sinaiticus, output in xml format, looks like this:
[040MAT.usx.txt](https://github.com/mimbrown/render/files/3047287/040MAT.usx.txt)

**Comment by mimbrown** [comment]
It's possible that this issue is caused by a discrepancy between versions. If so let me know. But I've run into this:
In the interlinear data, the words are arranged like this:
```
<Cluster>
  <Range Index="84" Length="3" />
  <Lexeme GlossId="Ey5Kfl7w" Id="Word:ΪΔΟ" />
</Cluster>
<Cluster>
  <Range Index="88" Length="1" />
  <Lexeme GlossId="A4T5vWpv" Id="Word:Υ" />
</Cluster>
```
However, in the Sinaiticus, these two words are joined together by an apostrophe: `ϊδο'υ`
In this situation, would you expect the two words to be preserved as two in the interlinear, with the apostrophe being included in at the beginning of the second word?

**Comment by mimbrown** [comment]
Also, I don't know the purpose of the `┬` character. Currently I'm stripping them out; do they need to remain? If so, how should they be resolved, since they aren't in the interlinear data?

**Comment by mimbrown** [comment]
Finally, there are instances where a period `.` is alone, not attached to a word. Does that need to be attached to the previous word?

**Comment by mimbrown** [comment]
You can look at the interlinear branch if you want to test this out.
Steps to get it working:
* Run `git checkout interlinear`.
* Rename or delete the `Interlinear.xml` file in the book of your choice (e.g. MAT).
* Add the Sinaiticus file in the book folder with the name `Sinaiticus.usx`.
* `./render` the book and validate the output.

**Comment by mnjames** [comment]
The words with <code>'</code> in them were actually a mistake in the encoding in Paratext. I've made the correction and they're now listed as single words and not broken. The one exception to this is in MAT 2:23, where the word <code>προφητω̅'</code> doesn't show up correctly in the Raw_Interlinear.xml data since the <code>'</code> is word-final.

The <code>┬</code> represents places where Sinaiticus was changed later. We need to keep that information in Paratext, but removing it during the printing process is exactly the right thing to do.

Yes, the problem is punctuation like <code>, . : ˙ ·</code>, which show up in the original files but not in the Raw_Interlinear.xml.

I tried the interlinear branch, but when it printed it just had blank lines for each of the Greek lines. It seems like both the Sinaiticus.usx information (the original text) and the Raw_Interlinear.xml data will be necessary--the 2nd because it's what connects each word to the lexicon entry, and without that the interlinearizer program will never create Interlinear.xml.

**Comment by mimbrown** [comment]
It sounds like it's using the old version of the `Interlinear.xml` file. Did you drop it so that it could be rebuilt? This is what my output looks like.
![Sample Interlinear](https://user-images.githubusercontent.com/8728868/55733328-dd2d8000-59da-11e9-9692-711844d24d1a.png)

**Comment by mnjames** [comment]
My mistake was removing the Raw_Interlinear.xml file. Now when I just remove the Interlinear.xml file I get this error:
```
Pre-processing files
Building the interlinear MAT
Discrepancy detected: Raw has more words
1:1 @ 1
lua: create-interlinear.lua:220: attempt to index a nil value (field '?')
stack traceback:
	create-interlinear.lua:220: in function 'buildInterlinear'
	create-interlinear.lua:251: in main chunk
	[C]: in ?
```

Note, I think with the changes I've made in the Paratext data regarding `'`, at least some of the code in `create-interlinear.lua` line 115
`for word in item:gsub("([^%s])'([^%s])", "%1 '%2"):gsub("%s*┬", ""):gmatch("[^%s]+") do`
is probably now irrelevant
[Raw_Interlinear.xml.txt](https://github.com/mimbrown/render/files/3057346/Raw_Interlinear.xml.txt)
[Sinaiticus.usx.txt](https://github.com/mimbrown/render/files/3057347/Sinaiticus.usx.txt)
[Lexicon.xml.txt](https://github.com/mimbrown/render/files/3057348/Lexicon.xml.txt)


.

**Comment by mimbrown** [comment]
That's strange. Does the Sinaiticus have any data in it? That error is telling you that there is a discrepancy between the number of words between the two documents in chapter 1 verse 1.

**Comment by mnjames** [comment]
I re-exported all the data and am not getting that error. I wonder if I exported the wrong files somehow last time.

I pushed a minor change because the code involving `'` was causing errors now that I've changed the way Paratext handles that character.

3 of the 6 books look great, but I'm getting this error in the other 3.
```
Pre-processing files
Building the interlinear MRK
lua: create-interlinear.lua:216: attempt to index a nil value (local 'verseTable')
stack traceback:
	create-interlinear.lua:216: in function 'buildInterlinear'
	create-interlinear.lua:251: in main chunk
	[C]: in ?
```

**Comment by mimbrown** [comment]
Could you post the files that are causing the error? I hope to be able to fix that today.

**Comment by mnjames** [comment]
[Raw_Interlinear.xml.txt](https://github.com/mimbrown/render/files/3069082/Raw_Interlinear.xml.txt)
[Sinaiticus.usx.txt](https://github.com/mimbrown/render/files/3069083/Sinaiticus.usx.txt)
[SSV.usx.txt](https://github.com/mimbrown/render/files/3069084/SSV.usx.txt)
[SSV_Lit.usx.txt](https://github.com/mimbrown/render/files/3069085/SSV_Lit.usx.txt)
[Lexicon.xml.txt](https://github.com/mimbrown/render/files/3069089/Lexicon.xml.txt)

**Comment by mimbrown** [comment]
There were a couple of issues. One was that I was not ignoring some extra stuff in the interlinear that should have been ignored. I fixed that. I also modified the builder so that it would at least print what the chapter and verse were that were causing issues. That output looks something like this:
```
Discrepancy detected: Sinaiticus is missing a verse
(See 12:20)
```
When I went to check out that verse in Sinaiticus, I found this:
`<verse number="20·" style="v" />`
The verse was there, but there was the extra `·` character in the number, which was the problem. You'll have to change that on your end as well. But Mark is working fine for me now.
